### PR TITLE
Report Buttons: View, Download + Run

### PR DESCRIPTION
**Pull request by gbdubs**
- Adds a shared set of buttons for viewing + dowloading a report. The download button uses `jsZip` which was the library with broadest usage that I found. Happy to supplant it for another. The zip downloads nicely, and can be opened (though the relative URLs leave the assets broken :( )
- Adds a `runAnalysis` button, with the goal being to use a standard hint in a few places to encourage folks to run audits, and to allow them to stay in the same place and visit the report once created. Works decently well, but ran into #148 , which will need independent attention
- Populates Snapshots in Analyses so we can see what they refer to.
- Populates ownerId on analysis so we can grey-out analyses that the user doesn't have access to (note this is not load-bearing authz, just used to prevent users from the frustration of an err message)
- Updates `getMaybeMe` to return back other info other than the `user`, in this case, the user's `ownerId`. This is more in line with what we do in AS, where the `me` call returns back announcements + personal tasks.
-  Fixes the `external` warning when navigating to view the report in browser
- Updates the `Metadata` sections of list views to just be an `always` standard debug until we can figure out what RMI wants out of that.
- Fixes the oapi spec's `blob_id` => `blobId`

### CODE REVIEW

File: frontend/components/analysis/AccessButtons.vue [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  await pactaClient.accessBlobContent({
+    items: props.analysis.artifacts.map((asset): AccessBlobContentReqItem => ({
+      blobId: asset.blob.id,
+    })),
+  }).then(async (response: AccessBlobContentResp) => {
+    const zip = new JSZip()
+    await Promise.all(response.items.map(
+      async (item): Promise<void> => {
+        const response = await fetch(item.downloadUrl)
+        const data = await response.blob()
```
Comment (bcspragu) [inline frontend/components/analysis/AccessButtons.vue:48]: There should be a way to stream the binary response data directly to the zip without loading it into memory first, can you see if that's an easy option?

File: frontend/components/analysis/AccessButtons.vue [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    const element = document.createElement('a')
+    element.href = URL.createObjectURL(content)
+    const fileName = `${props.analysis.name}.zip`
+    element.download = fileName
+    document.body.appendChild(element)
+    element.click()
+    document.body.removeChild(element)
+    downloadInProgress.value = false
+  })
+}
```
Comment (bcspragu) [inline frontend/components/analysis/AccessButtons.vue:65]: Mixing async/await and promises is never a good idea, one is just syntactic sugar for the other. I'd just commit to async/await, e.g.

```suggestion
const doDownload = async () => {
  downloadInProgress.value = true
  const response = await pactaClient.accessBlobContent({
    items: props.analysis.artifacts.map((asset): AccessBlobContentReqItem => ({
      blobId: asset.blob.id,
    })),
  })

  const zip = new JSZip()
  await Promise.all(response.items.map(
    async (item): Promise<void> => {
      const response = await fetch(item.downloadUrl)
      const data = await response.blob()
      const blob = presentOrFileBug(props.analysis.artifacts.find((artifact) => artifact.blob.id === item.blobId)).blob
      const fileName = `${blob.fileName}`
      zip.file(fileName, data)
    }),
  )
  const content = await zip.generateAsync({ type: 'blob' })

  const element = document.createElement('a')
  element.href = URL.createObjectURL(content)
  const fileName = `${props.analysis.name}.zip`
  element.download = fileName
  document.body.appendChild(element)
  element.click()
  document.body.removeChild(element)
  downloadInProgress.value = false
}
```

**Review by bcspragu — APPROVED** [review]

File: frontend/components/analysis/AccessButtons.vue (hunk 1 above)
Comment (gbdubs) [reply frontend/components/analysis/AccessButtons.vue:48]: I played around with a few attempts here, without avail. The docs are here, nothing screams out.

https://stuk.github.io/jszip/documentation/api_jszip/file_data.html

File: frontend/components/analysis/AccessButtons.vue (hunk 2 above)
Comment (gbdubs) [reply frontend/components/analysis/AccessButtons.vue:65]: SG. Done.

File: frontend/components/analysis/AccessButtons.vue (hunk 1 above)
Comment (bcspragu) [reply frontend/components/analysis/AccessButtons.vue]: Took a look, looks like streaming is supported in Node environments, but not in the browser, c'est la vie

**Pull request merged**